src/main/java/com/sap/charging/server/api/v1/store/OptimizerSettings.java

```diff
@@ -0,0 +1,79 @@
+package com.sap.charging.server.api.v1.store;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class OptimizerSettings {
+
+    private double weightObjectiveFairShare;
+    private double weightObjectivePeakShaving;
+    private double weightObjectiveEnergyCosts;
+    private double weightObjectiveLoadImbalance;
+
+    public OptimizerSettings() {}
+
+    @JsonCreator
+    public OptimizerSettings(@JsonProperty("weightObjectiveFairShare") double weightObjectiveFairShare,
+        @JsonProperty("weightObjectivePeakShaving") double weightObjectivePeakShaving,
+        @JsonProperty("weightObjectiveEnergyCosts") double weightObjectiveEnergyCosts,
+        @JsonProperty("weightObjectiveLoadImbalance") double weightObjectiveLoadImbalance) {
+        this.weightObjectiveFairShare = weightObjectiveFairShare;
+        this.weightObjectivePeakShaving = weightObjectivePeakShaving;
+        this.weightObjectiveEnergyCosts = weightObjectiveEnergyCosts;
+        this.weightObjectiveLoadImbalance = weightObjectiveLoadImbalance;
```

Comment:
Does the weight values have a min and max (min is zero I guess) values to check?
I think it should worse the effort to add some sanity checks regarding the values inputed.

Reply:
The real-time heuristic doesn't really use the weights (for reference see [here](https://github.com/SAP/emobility-smart-charging/blob/master/src/main/java/com/sap/charging/realTime/StrategyAlgorithmic.java#L91). The weights are used to determine the most important objective. The most important objective is used to schedule EVs above their minimum SoC. 

The weights are used numerically in the offline optimization approach (for reference see [here](https://github.com/SAP/emobility-smart-charging/blob/master/src/main/java/com/sap/charging/opt/lp/InstanceLP.java)) and don't really have a maximum or minimum since they are simply used in the linear programming model...

However, I agree that in the REST API (which only exposes the real-time heuristic) negative weights don't make sense. I'll add sanity checks to this class.